src/deploy/missions/contracts/z-token/z-token.ts

```diff
@@ -86,35 +86,36 @@ IZNSContracts
   }
 
   getArtifact () {
-    return meowArtifact;
+    return zArtifact;
   }
 
+  // it will choose the governon as `admin` argument
+  // and deployAdmin as `minter` and first passed admin as `mintBeneficiary`.
   async deployArgs () : Promise<TDeployArgs> {
-    return [meowTokenName, meowTokenSymbol];
+    const {
+      initialAdminDelay,
+      initialSupplyBase,
+      inflationRates,
+      finalInflationRate,
+    } = this.config.zTokenConfig as IZTokenConfig;
+    return [
+      zTokenName,
+      zTokenSymbol,
+      this.config.governorAddresses[0],
```

Comment:
destructure this properly as you did with other values. also, these specific role based addresses should be part of the config for zToken